File: source codes/Lab 12/ArrayMethods.cs
using System;
using System.Collections.Generic;
using System.Linq;

class MainClass {
  public static int ArrayMax(int[] x){
    return x.Max();
  }
  public static int ArrayMin(int[] x){
    return x.Min();
  }
  public static void ArraySquared(int[] x){
    for (int j = 0;j < x.Length;j++) {
      x[j] = x[j]*x[j];
      Console.Write((x[j])+ ", ");
    }
  }
  public static void ArrayReverse(int[] y){
    Array.Reverse(y);
    for (int j = 0;j < y.Length;j++)
      {Console.Write(y[j]+ ", ");}
  }
  public static void Main() {
    int size,count=0,test;
    Console.WriteLine("What is the array size: ");
    size = Convert.ToInt32(Console.ReadLine());
    int[] Arr1 = new int[size];
    Console.WriteLine("Enter the numbers for the array: ");
    while (count < size){
      test = Convert.ToInt32(Console.ReadLine());
      Arr1[count] = test;
      count++;
    }
    Console.Write("Original Array: ".PadRight(20));
    for (int k = 0;k < Arr1.Length;k++)
      {Console.Write(Arr1[k] + ", ");}
    Console.WriteLine();
    Console.WriteLine("Array Miximum: ".PadRight(20) + ArrayMax(Arr1));
    Console.WriteLine("Array Minimum: ".PadRight(20) + ArrayMin(Arr1));
    Console.Write("Array Squared: ".PadRight(20));
    ArraySquared(Arr1);
    Console.WriteLine();
    Console.Write("Array Reversed: ".PadRight(20));
    ArrayReverse(Arr1);
  }
}

File: School source codes/Lab 12/ArrayMethods.rb
def ArrayMax(x)
  x.max
end
def ArrayMin(x)
  x.min
end
def ArraySquared(x)
  x.map! {|n| n **2}
end
def ArrayReverse(x)
  x.reverse!
end
count = 0
puts "What is the array size: "
size = gets.chomp.to_i
Arr1 = Array.new(size)
puts "Enter the numbers for the array: "
while (count < size)
  test = gets.chomp.to_i
  Arr1[count] = test
  count += 1
end
print "Original Array: "
print Arr1.join(' ')
puts "\n"
puts "Array Miximum: #{ArrayMax(Arr1)}"
puts "Array Minimum: #{ArrayMin(Arr1)}"
print "Array Squared: "
ArraySquared(Arr1)
print Arr1.join(" ")
puts "\n"
print "Array Reversed: "
ArrayReverse(Arr1)
print Arr1.join(" ")